src/frontend/src/pages/ProjectDetailPage/ProjectViewContainer/TaskListTabPanel.tsx

```diff
@@ -3,28 +3,201 @@
+        open={Boolean(anchorEl)}
+        onClose={handleClose}
+      >
+        <MenuItem disabled>
+          <ListItemIcon>
+            <DeleteIcon fontSize="small" />
+          </ListItemIcon>
+          Delete
+        </MenuItem>
+        {status !== TaskStatus.IN_BACKLOG && (
+          <MenuItem onClick={handleClose} disabled={disabled}>
+            <ListItemIcon>
+              <PauseIcon fontSize="small" />
+            </ListItemIcon>
+            Move to Backlog
+          </MenuItem>
+        )}
+        {status !== TaskStatus.IN_PROGRESS && (
+          <MenuItem onClick={handleClose} disabled={disabled}>
+            <ListItemIcon>
+              <PlayArrowIcon fontSize="small" />
+            </ListItemIcon>
+            Move to In Progress
+          </MenuItem>
+        )}
+        {status !== TaskStatus.DONE && (
+          <MenuItem onClick={handleClose} disabled={disabled}>
+            <ListItemIcon>
+              <CheckIcon fontSize="small" />
+            </ListItemIcon>
+            Move to Done
+          </MenuItem>
+        )}
+      </Menu>
+    </>
+  );
+
+  const renderNotes = () => <Link>See Notes</Link>;
+
+  const renderHeader = (params: GridColumnHeaderParams) => (
+    <Typography sx={{ fontWeight: 'bold' }}>{params.colDef.headerName}</Typography>
+  );
+
+  const baseColDef: GridColDefStyle = {
+    flex: 1,
+    align: 'center',
+    headerAlign: 'center'
+  };
+
+  const columns: GridColDef[] = [
+    { ...baseColDef, field: 'title', headerName: 'Title', width: 90, renderHeader: renderHeader },
+    {
+      ...baseColDef,
+      field: 'notes',
+      headerName: 'Notes',
+      renderCell: renderNotes,
+      width: 150,
+      renderHeader: renderHeader
+    },
+    {
+      ...baseColDef,
+      field: 'deadline',
+      headerName: 'Deadline',
+      width: 150,
+      renderHeader: renderHeader
+    },
+    {
+      ...baseColDef,
+      field: 'priority',
+      headerName: 'Priority',
+      width: 150,
+      renderHeader: renderHeader
+    },
+    {
+      ...baseColDef,
+      field: 'assignee',
+      headerName: 'Assignee',
+      width: 150,
+      renderHeader: renderHeader
+    },
+    {
+      ...baseColDef,
+      field: 'actions',
+      headerName: 'Actions',
+      renderCell: renderActionButtons,
+      width: 150,
+      renderHeader: renderHeader
+    }
+  ];
+  const rows = tasks.map((task: Task, idx: number) => {
+    const date = new Date(task.deadline);
+    const assigneeString = task.assignees.reduce(
+      (accumulator: string, currentVal: UserPreview) => accumulator + namePipe(currentVal),
+      ''
+    );
+    return {
+      id: idx,
+      title: task.title,
+      deadline: datePipe(date),
+      priority: task.priority,
+      assignee: assigneeString.substring(0, assigneeString.length - 2)
+    };
+  });
+
   return (
     <div role="tabpanel" hidden={value !== index} id={`simple-tabpanel-${index}`} aria-labelledby={`simple-tab-${index}`}>
```

Comment:
what does index represent? and why is it hidden if value equals index?

Reply:
oh is it the currently selected tab panel? So we hide the other ones if the selected index doesnt equal what the value of the tabpanel is on render? I feel like we should probably do this a seperate way if thats the case. I may be misunderstanding the code

Reply:
yeah, I think was how they did it in the example, but tbh you'd have to ask @jonathan-chen10 he was the one who wrote this bit of the code

Reply:
yeah i copied some code from the mui docs and didn't quite figure out how to refactor it to be cleaner. tabs are 0, 1, 2 iirc and if you're not on the right tab it hides the contents. i tried making the indices word-based but i don't remember why i gave up on that